File: purchasely/ios/PurchaselyViewManager.m
//
//  NativeViewManager.m
//  example
//
//  Created by Chouaib Mounaime on 05/03/2024.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_REMAP_MODULE(PurchaselyView, PurchaselyViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(placementId, NSString)
RCT_EXPORT_VIEW_PROPERTY(presentation, NSDictionary)

RCT_EXTERN_METHOD(onPresentationClosed:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject);

@end

File: packages/purchasely/ios/PurchaselyViewManager.swift
//
//  NativeViewManager.swift
//  example
//
//  Created by Chouaib Mounaime on 05/03/2024.
//

import Foundation

@objc (PurchaselyViewManager)
class PurchaselyViewManager: RCTViewManager {

  // MUST stay false. Fabric instantiates this manager's view on the main
  // thread; returning true would force any `NativeModules.PurchaselyView`
  // resolution to `dispatch_sync` onto that same main queue, deadlocking if it
  // happens while the view is mid-creation. Returning false lets each thread
  // resolve its own instance without a cross-thread hop. `view()` is still
  // called on the main queue by UIManager regardless, and init does no UIKit
  // work.
  override static func requiresMainQueueSetup() -> Bool {
    return false
  }

  override func view() -> UIView! {
    return PurchaselyView()
  }
}
